libs/services/license/src/lib/barcode.service.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  LicenseType,
+  LicenseVerifyExtraDataResult,
+} from '@island.is/clients/license-client'
+import { ConfigType } from '@nestjs/config'
+import { Inject, Injectable } from '@nestjs/common'
+import { Cache as CacheManager } from 'cache-manager'
+import { sign, SignOptions, verify } from 'jsonwebtoken'
+import { LICENSE_SERVICE_CACHE_MANAGER_PROVIDER } from './licenseCache.provider'
+import { LicenseConfig } from './license.config'
+
+const BARCODE_EXPIRE_TIME_IN_SEC = 60
+
+/**
+ * License token data used to generate a license token
+ * The reason for the one letter fields is to keep the token as small as possible, since it will be used to generate barcodes
+ */
+export type LicenseTokenData = {
+  /**
+   * Version
+   */
+  v: string
+  /**
+   * License type
+   */
+  t: string
+  /**
+   * Code (Reference to redis record with license data)
+   */
+  c: string
+}
+
+export type BarcodeData<Type extends LicenseType> = {
+  nationalId: string
+  licenseType: Type
+  extraData: LicenseVerifyExtraDataResult<Type>
+}
+
+@Injectable()
+export class BarcodeService {
+  constructor(
+    @Inject(LicenseConfig.KEY)
+    private readonly config: ConfigType<typeof LicenseConfig>,
+    @Inject(LICENSE_SERVICE_CACHE_MANAGER_PROVIDER)
+    private readonly cacheManager: CacheManager,
+  ) {}
+
+  public tokenExpiredError = 'TokenExpiredError'
```

Comment:
Should not store constants in member variables. Just export it as `TOKEN_EXPIRED_ERROR` or something :)